pyomo/repn/plugins/gams_writer.py

```diff
@@ -143,56 +231,79 @@ def _print_model( self, model, output_file, solver_capability, var_list,
         for con in model.component_data_objects(Constraint, active=True):
             if con.body.is_fixed():
                 continue
+
+            # Ensure GAMS will not encounter domain violations in presolver
+            # operations at current values, which are None (0) by default
+            # Used to handle log and log10 violations, for example
+            try:
+                value(con.body)
+            except:
+                raise RuntimeError("While evaluating: %s\n\tProblemWriter_gams"
+                                   " encountered an error. Ensure initial"
+                                   " variable values\n\tdo not violate"
+                                   " any domains (are you using log/log10?)"
+                                 % con.name)
```

Comment:
As a general rule, the following is bad practice:

```
try:
   # .....
except:
  raise <something else>

```
for two reasons:
  1. "blind" `except` clauses will catch all manner of exceptions, not just the one you are expecting.  In this case, you should probably only trap a `ValueError`
  2. Swallowing the exception and raising something new destroys the original message that was generated when throwing the exception, along with the entire stack trace below this point (which makes debugging *hard*).

If you are going to do this, you should use a logger to emit your message and reraise the original exception.